keps/sig-apps/3329-retriable-and-non-retriable-failures/kep.yaml

```diff
@@ -0,0 +1,45 @@
+title: Retriable and non-retriable Pod failures for Jobs
+kep-number: 3329
+authors:
+  - "@mimowo"
+owning-sig: sig-apps
+participating-sigs:
+  - sig-scheduling
+  - sig-api-machinery
+status: implementable
+creation-date: 2022-06-07
+reviewers:
+  - "@liggitt"
+approvers:
+  - "@soltysh"
+  - "@alculquicondor"
+  - "@deads2k"
+
+# The target maturity stage in the current dev cycle for this KEP.
+stage: alpha
+
+# The most recent milestone for which work toward delivery of this KEP has been
+# done. This can be the current (upcoming) milestone, if it is being actively
+# worked on.
+latest-milestone: "v1.25"
+
+# The milestone at which this feature was, or is targeted to be, at each stage.
+milestone:
+  alpha: "v1.25"
+  beta: "v1.26"
+  stable: "v1.27"
+
+# The following PRR answers are required at alpha release
+# List the feature gate name and the components for which it must be enabled
+feature-gates:
+  - name: JobBackoffPolicy
+    components:
+      - kube-apiserver
+      - kube-controller-manager
+      - kube-scheduler
+      - kubelet
+disable-supported: true
+
+# The following PRR answers are required at beta release
+metrics:
```

Comment:
I'd suggest thinking about some metrics around backoffs and describe various approaches. They will be required for beta, so you have some time to play with different options as long as it's alpha.

Reply:
I think we can add 2 metrics:

- job_terminated_total, with a label indicating whether it failed due to a Termination rule or reaching backoff or activeDeadline. Or we could add a reason label to the existing job_finished_total
- job_pod_failure_total with a label that matches the action jobTerminated, skipped. Or just job_pod_failure_skipped_total?